test/fixtures/payload-helpers/features/compare_requests_json_fixtures.feature

```diff
@@ -0,0 +1,47 @@
+Feature: Testing helper methods respond correctly
+
+    Scenario: The request body matches an expected error payload
+        When I send a "payload"-type request
+        Then I wait to receive a request
+        And the request is valid for the error reporting API version "4.0" for the "Maze-runner" notifier
+
+    Scenario: The request body matches an expected session payload
+        When I send a "session"-type request
+        Then I wait to receive a request
+        And the request is valid for the session reporting API version "1.0" for the "Maze-runner" notifier
+
+    Scenario: The request body is correct for an unhandled payload
+        When I send a "unhandled"-type request
+        Then I wait to receive a request
+        And event 0 is unhandled
+
+    Scenario: The request body is correct for an unhandled payload with session data
+        When I send an "unhandled session"-type request
```

Comment:
Minor nit, but it may be worth tweaking the name as it sounds like the session has the unhandled state rather than the error